File: src/main/java/com/kamikazejam/datastore/mode/profile/store/ProfileStorageDatabase.java
package com.kamikazejam.datastore.mode.profile.store;

import com.kamikazejam.datastore.base.Cache;
import com.kamikazejam.datastore.base.storage.StorageDatabase;
import com.kamikazejam.datastore.mode.profile.StoreProfile;
import org.jetbrains.annotations.NotNull;

import java.util.UUID;

public class ProfileStorageDatabase<X extends StoreProfile> extends StorageDatabase<UUID, X> {

    public ProfileStorageDatabase(Cache<UUID, X> cache) {
        super(cache);
    }

    @Override
    public @NotNull String getLayerName() {
        return "Profile MongoDB";
    }
}

File: src/main/kotlin/com/kamikazejam/datastore/mode/profile/store/ProfileStorageDatabase.kt
package com.kamikazejam.datastore.mode.profile.store

import com.kamikazejam.datastore.base.Collection
import com.kamikazejam.datastore.base.storage.StorageDatabase
import com.kamikazejam.datastore.mode.profile.StoreProfile
import java.util.*

class ProfileStorageDatabase<X : StoreProfile<X>>(collection: Collection<UUID, X>) :
    StorageDatabase<UUID, X>(collection) {
    override val layerName: String
        get() = "Profile MongoDB"
}
